content/running_bookwyrm/cli.md

```diff
@@ -0,0 +1,143 @@
+Title: Command Line Tool
+Date: 2021-11-06
+Order: 6
+
+Bookwyrm developers and instance managers can use the `bw-dev` script for common tasks. This can make your commands shorter, easier to remember, and harder mess up.
+
+Once you have installed Bookwyrm [in production](installing-in-production.html) or [in development](https://docs.joinbookwyrm.com/developer-environment.html#setting_up_the_developer_environment), you can run the script from the command line with `./bw-dev` followed by the subcommand you want to run.
+
+## Docker shortcuts
+
+### bash
+
+Open an interactive `bash` session inside the docker `web` container.
+
+### build
+
+Equivalent to `docker-compose build`.
+
+### dbshell
+
+Open an interactive Postgres database shell. I hope you know what you're doing.
+
+### run
+
+Run an arbitrary command in the `web` cointainer.
+
+Equivalent to `docker-compose run --rm --service-ports web`. 
```

Comment:
I use this when I need to use a `pdb` on my local development environment (I stop the regular `web` worker and then run this). It's terribly named; maybe something like `service-ports-web` would be better?

Reply:
I wasn't previously familiar with this flag - what's the reason to use `--service-ports`? Is it because you expect the container to not actually be running (given you're using this for testing purposes)?

Reply:
If you run the container normally, using a pdb statement will cause an error (presumably so that you don't accidentally let random people run arbitrary code), the service-ports flag allows the container to expose the pdb prompt, so you can put a breakpoint in and actually use it.